apps/client/src/features/modals/ViewsSettingsModal.jsx

```diff
@@ -75,74 +82,109 @@ export default function ViewsSettingsModal() {
-              />
-            </FormControl>
+        <hr className={styles.divider} />
+        <div className={style.hSeparator}>Timer Options</div>
+        <div className={styles.splitSection}>
+          <div>
+            <span className={`${styles.sectionTitle} ${styles.main}`}>Normal Color</span>
+            <span className={styles.sectionSubtitle}>Change timer Normal Color</span>
+          </div>
+          <PopoverPicker
+            name='normalColor'
+            color={formData.normalColor}
+            onChange={(event) => handleChange('normalColor', event)}
+          />
+        </div>
+        <div className={styles.splitSection}>
+          <div>
+            <span className={`${styles.sectionTitle} ${styles.main}`}>Warning Color</span>
+            <span className={styles.sectionSubtitle}>Change timer Warning Color</span>
           </div>
+          <PopoverPicker
+            name='warningColor'
+            color={formData.warningColor}
+            onChange={(event) => handleChange('warningColor', event)}
+          />
+        </div>
+        <FormControl className={styles.splitSection}>
+          <label htmlFor='warningThreshold'>
+            <span className={styles.sectionTitle}>Warning Time</span>
+            <span className={styles.sectionSubtitle}>The time (in minutes) when the color changes</span>
+          </label>
+          <InputGroup size='sm' width='140px'>
+            <NumberInput
+              {...numberInputProps}
+              id='warningThreshold'
+              variant='ontime-filled-on-light'
+              value={millisToMinutes(Number(formData.warningThreshold), 'm')}
+              onChange={(event) => handleThresholdChange('warningThreshold', event)}
+            >
+              <NumberInputField />
+            </NumberInput>
+            <InputRightAddon children='Minutes'></InputRightAddon>
+          </InputGroup>
+        </FormControl>
+        <div className={styles.splitSection}>
+          <div>
+            <span className={`${styles.sectionTitle} ${styles.main}`}>Danger Color</span>
+            <span className={styles.sectionSubtitle}>Change timer Danger Color</span>
+          </div>
+          <PopoverPicker
+            name='dangerColor'
+            color={formData.dangerColor}
+            onChange={(event) => handleChange('dangerColor', event)}
+          />
+        </div>
+        <FormControl className={styles.splitSection}>
+          <label htmlFor='dangerThreshold'>
+            <span className={styles.sectionTitle}>Danger Time</span>
+            <span className={styles.sectionSubtitle}>The time (in minutes) when the color changes</span>
+          </label>
+          <InputGroup size='sm' width='140px'>
+            <NumberInput
+              {...numberInputProps}
+              id='dangerThreshold'
+              variant='ontime-filled-on-light'
+              value={millisToMinutes(Number(formData.dangerThreshold), 'm')}
```

Comment:
this sorts of calculations should probably happen before the return, just for clarity